project/profiles/capacity4more/modules/c4m/og/c4m_og_group_menu/c4m_og_group_menu.module

```diff
@@ -95,3 +95,72 @@ function c4m_og_group_menu_page_access() {
 
   return og_menu_access('node', $gid);
 }
+
+/**
+ * Implements hook_form_FORM_ID_alter().
+ */
+function c4m_og_group_menu_form_og_menu_edit_item_form_alter(&$form, &$form_state, $form_id) {
+  $gid = og_menu_get_context();
+  $group = node_load($gid);
+  if (!$group || $group->type != 'project') {
+    return;
+  }
+
+  // The select element of the wiki pages should not be displayed on a default
+  // menu link.
+  if ($form['link_path']['#type'] == 'value') {
+    return;
+  }
+
+  $tree = menu_build_tree($group->book['menu_name']);
+  // Resolve pages under the node of the menu tree.
+  $top_pages = array_shift($tree)['below'];
+
+  // Default select option.
+  $wiki_details = array('' => t('--- SELECT ---'));
+
+  // Generate page details for each page at tree.
+  foreach ($top_pages as $top_page) {
+    _c4m_og_group_menu_process_page($top_page, $wiki_details);
+  }
+
+  // Add select element to form.
+  $form['wiki_select'] = array(
+    '#title' => t('Wiki Page'),
+    '#type' => 'select',
+    '#options' => $wiki_details,
+    '#weight' => -1,
+  );
+
+  // If path provided, set selected Wiki Page as a default value.
+  if (in_array($form['link_path']['#default_value'], array_keys($wiki_details))) {
+    $form['wiki_select']['#default_value'] = $form['link_path']['#default_value'];
+  }
+
+  $form['link_title']['#weight'] = -3;
+  $form['link_path']['#weight'] = -2;
+
+  drupal_add_js(drupal_get_path('module', 'c4m_og_group_menu') . '/js/wiki-select.js');
+}
+
+/**
+ * Recursive function.
```

Comment:
This should be part of the explanation of the function, please move it down and elaborate the logic.